File: bank-service/src/main/java/rs/edu/raf/repository/CurrencyRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.Currency;

import java.util.Optional;

@Repository
public interface CurrencyRepository extends JpaRepository<Currency, Long> {
    Optional<Currency> findByCode(String code);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/CurrencyRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.Currency
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CurrencyRepository: JpaRepository<Currency, Long> {
    fun findByCode(code: String): Optional<Currency>
}